scripts/ci/verify-adoption-doc.py

```
@@ -0,0 +1,204 @@
+            continue
+
+        h2 = latest_h2_before(hm.start())
+        verify_header = section_text.split("\n", 1)[0].strip()
+        blocks.append((h2, verify_header, code))
+
+    return blocks
+
+
+def run_block(code: str, cwd: Path) -> Tuple[int, str]:
+    """Execute a bash block, return (exit_code, combined_output)."""
+    proc = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)